client/client.go

```diff
@@ -142,152 +165,7 @@ func (r *Request) SetProxy(proxyUrl string) error {
 	return nil
 }
 
+// Deprecated: Headers shouldn't be modified after construction. Use WithExtraHeaders instead
 func (r *Request) AddHeader(key, value string) {
 	r.Headers[key] = value
 }
-
-func (r *Request) GetWithContext(ctx context.Context, result interface{}, path string, query url.Values) error {
-	uri := r.GetURL(path, query)
-	return r.Execute(ctx, "GET", uri, nil, result)
-}
-
-func (r *Request) Get(result interface{}, path string, query url.Values) error {
-	return r.GetWithContext(context.Background(), result, path, query)
-}
-
-func (r *Request) Post(result interface{}, path string, body interface{}) error {
-	return r.PostWithContext(context.Background(), result, path, body)
-}
-
-func (r *Request) GetRaw(path string, query url.Values) ([]byte, error) {
-	uri := r.GetURL(path, query)
-	return r.ExecuteRaw(context.Background(), "GET", uri, nil)
-}
-
-func (r *Request) PostRaw(path string, body interface{}) ([]byte, error) {
-	buf, err := GetBody(body)
-	if err != nil {
-		return nil, err
-	}
-	uri := r.GetBase(path)
-
-	return r.ExecuteRaw(context.Background(), "POST", uri, buf)
-}
-
-func (r *Request) PostWithContext(ctx context.Context, result interface{}, path string, body interface{}) error {
-	buf, err := GetBody(body)
-	if err != nil {
-		return err
-	}
-	uri := r.GetBase(path)
-	return r.Execute(ctx, "POST", uri, buf, result)
-}
-
-func (r *Request) Execute(ctx context.Context, method string, url string, body io.Reader, result interface{}) error {
-	req, err := http.NewRequest(method, url, body)
-	if err != nil {
-		return err
-	}
-
-	for key, value := range r.Headers {
-		req.Header.Set(key, value)
-	}
-
-	b, err := r.execute(ctx, req)
-	if err != nil {
-		return err
-	}
-
-	err = json.Unmarshal(b, result)
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-func (r *Request) ExecuteRaw(ctx context.Context, method string, url string, body io.Reader) ([]byte, error) {
-	req, err := http.NewRequest(method, url, body)
-	if err != nil {
-		return nil, err
-	}
-
-	for key, value := range r.Headers {
-		req.Header.Set(key, value)
-	}
-
-	return r.execute(ctx, req)
-}
-
-func (r *Request) execute(ctx context.Context, req *http.Request) ([]byte, error) {
-	c := r.HttpClient
-
-	startTime := time.Now()
-	res, err := c.Do(req.WithContext(ctx))
-
-	if r.metricsEnabled() {
-		r.httpMetrics.observeDuration(req, startTime)
-		r.httpMetrics.observeResult(req, res, err)
-	}
-
-	if err != nil {
-		return nil, err
-	}
-
-	err = r.HttpErrorHandler(res, req.URL.String())
-	if err != nil {
-		return nil, err
-	}
-
-	if res.StatusCode < http.StatusOK || res.StatusCode >= http.StatusBadRequest {
-		defer res.Body.Close()
-		body, err := io.ReadAll(res.Body)
-		if err != nil {
-			return nil, err
-		}
```

Comment:
These lines are duplicated with [these](https://github.com/trustwallet/go-libs/pull/177/files#diff-bd3a55a72186f59e2e63efb4951573b2f9e4a7cc98086e922b0859f8ccc1dd09L256-L260). In the new execute function, they are moved above to eliminate duplication